### ISSUE TITLE
Queries are logged twice when using a STDOUT logger in the Rails 4 console

### ISSUE THREAD

**Issue by davidbalbert** [rails/rails#11415]
I've found what I believe to be a regression in Active Record's logging behavior inside the Rails console in Rails 4. If I set `config.logger` to log to STDOUT, all the database queries logged to the console get logged twice.

Here are the steps to reproduce:

```
$ rails --version
Rails 4.0.0
$ rails new test_logger
$ cd test_logger

Edit config/environments/development.rb and add these two lines in the configure block:

config.logger = ActiveSupport::Logger.new(STDOUT)
config.logger.level = ActiveSupport::Logger.const_get(ENV['LOG_LEVEL'] ? ENV['LOG_LEVEL'].upcase : 'DEBUG')

$ rails g model User name:string
$ rake db:migrate

$ rails c
Loading development environment (Rails 4.0.0)
2.0.0-p247 :001 > User.first
  User Load (0.2ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT 1
  User Load (0.2ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT 1
 => nil 
2.0.0-p247 :002 > 
```

I expect the "User Load" line to appear only once, but it appears twice. I can reproduce the bug both in Rails 4.0.0 and the latest master at the time of this writing. Following the same steps using Rails 3.2.13 (substituting `Logger` for `ActiveSupport::Logger`), the console works as expected.

I have also tried using `Logger` with the Rails 4 app. When I do this, the query is still logged twice, once using Ruby's default `Formatter` (as I would expect) and once using Rails's custom `Formatter` (unexpected):

```
rails c
Loading development environment (Rails 4.0.0)
2.0.0-p247 :001 > User.first
  User Load (0.3ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT 1
D, [2013-07-12T16:29:59.363272 #22484] DEBUG -- :   User Load (0.3ms)  SELECT "users".* FROM "users" ORDER BY "users"."id" ASC LIMIT 1
 => nil 
2.0.0-p247 :002 > 
```

Please let me know if there's anything else I can do to help with this bug.

**Comment by seanahrens** [comment]
For anyone who stumbles upon this issue from google, make sure that what's actually happening to you is not that your app is actually making duplicate REQUESTS as mine was. See http://stackoverflow.com/questions/3266020/rails-app-logging-duplicate-requests/28299100#28299100

**Comment by hakunin** [comment]
Still happening on rails 4.1.10

**Comment by benjaminb10** [comment]
And still in 4.2.1

**Comment by dwbutler** [comment]
I ended up adding this to application.rb:

```ruby
console do
  ActiveRecord::Base.logger = Rails.logger = Logger.new(STDOUT)
end
```

**Comment by micahlmartin** [comment]
@dwbutler That fixed it for me. I was still experiencing the issue in `4.2.3`.

**Comment by philbattos** [comment]
I was seeing duplicate logs in development but i discovered that it was caused by these two gems: ```rails_12factor``` and ```rails_stdout_logging```. I added them based on [Heroku's recommendation](https://devcenter.heroku.com/articles/ruby-support#injected-plugins). When i removed them, everything worked great. (I'm going to keep using the gems but i thought this info might help someone.)

**Comment by raldred** [comment]
Still present in fresh 4.2.4 app created today.

**Comment by hosh** [comment]
Same here

**Comment by nwjsmith** [comment]
In [my PR][1] I've added an option to the logger to disable broadcasting messages to other loggers, so you'll be able to do the following in `config/application.rb`:

``` ruby
config.logger = ActiveSupport::Logger.new(STDOUT)
config.logger.level = ActiveSupport::Logger.const_get(ENV['LOG_LEVEL'] ? ENV['LOG_LEVEL'].upcase : 'DEBUG')
config.logger.broadcast_messages = false
```

and the log will no longer print to stdout twice.

[1]: https://github.com/rails/rails/pull/22592

**Comment by nwjsmith** [comment]
This has been fixed with https://github.com/rails/rails/pull/22592

**Comment by dammer** [comment]
@nwjsmith
After update on 5.0.0.beta1 have this error on puma/webrick: 
```ruby
[2015-12-21 02:20:40] INFO  WEBrick 1.3.1
[2015-12-21 02:20:40] INFO  ruby 2.2.2 (2015-04-13) [x86_64-linux]
[2015-12-21 02:20:40] INFO  WEBrick::HTTPServer#start: pid=14710 port=3000
[2015-12-21 02:20:43] ERROR NameError: undefined local variable or method `broadcast_messages' for #<RailsStdoutLogging::StdoutLogger:0x00558acf7bdec8>
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/logger.rb:14:in `block (2 levels) in broadcast'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/logger.rb:400:in `debug'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/rack/logger.rb:35:in `call_app'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/rack/logger.rb:24:in `block in call'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/tagged_logging.rb:70:in `block in tagged'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/tagged_logging.rb:26:in `tagged'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/tagged_logging.rb:70:in `tagged'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/rack/logger.rb:24:in `call'
	/home/damir/project/.bundle/gems/gems/actionpack-5.0.0.beta1/lib/action_dispatch/middleware/request_id.rb:24:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/method_override.rb:22:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/runtime.rb:22:in `call'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
	/home/damir/project/.bundle/gems/gems/actionpack-5.0.0.beta1/lib/action_dispatch/middleware/load_interlock.rb:13:in `call'
	/home/damir/project/.bundle/gems/gems/actionpack-5.0.0.beta1/lib/action_dispatch/middleware/static.rb:132:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/sendfile.rb:111:in `call'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/engine.rb:522:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/handler/webrick.rb:86:in `service'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/webrick/httpserver.rb:138:in `service'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/webrick/httpserver.rb:94:in `run'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/webrick/server.rb:294:in `block in start_thread'
[2015-12-21 02:24:29] ERROR NameError: undefined local variable or method `broadcast_messages' for #<RailsStdoutLogging::StdoutLogger:0x00558acf7bdec8>
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/logger.rb:14:in `block (2 levels) in broadcast'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/logger.rb:400:in `debug'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/rack/logger.rb:35:in `call_app'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/rack/logger.rb:24:in `block in call'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/tagged_logging.rb:70:in `block in tagged'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/tagged_logging.rb:26:in `tagged'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/tagged_logging.rb:70:in `tagged'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/rack/logger.rb:24:in `call'
	/home/damir/project/.bundle/gems/gems/actionpack-5.0.0.beta1/lib/action_dispatch/middleware/request_id.rb:24:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/method_override.rb:22:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/runtime.rb:22:in `call'
	/home/damir/project/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/cache/strategy/local_cache_middleware.rb:28:in `call'
	/home/damir/project/.bundle/gems/gems/actionpack-5.0.0.beta1/lib/action_dispatch/middleware/load_interlock.rb:13:in `call'
	/home/damir/project/.bundle/gems/gems/actionpack-5.0.0.beta1/lib/action_dispatch/middleware/static.rb:132:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/sendfile.rb:111:in `call'
	/home/damir/project/.bundle/gems/gems/railties-5.0.0.beta1/lib/rails/engine.rb:522:in `call'
	/home/damir/project/.bundle/gems/bundler/gems/rack-f4562619c3c6/lib/rack/handler/webrick.rb:86:in `service'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/webrick/httpserver.rb:138:in `service'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/webrick/httpserver.rb:94:in `run'
	/home/damir/.rbenv/versions/2.2.2/lib/ruby/2.2.0/webrick/server.rb:294:in `block in start_thread'

```
using debugger i see 

```ruby 
[10, 19] in /home/damir/projects/seomix/.bundle/gems/gems/activesupport-5.0.0.beta1/lib/active_support/logger.rb
   10:     # Broadcasts logs to multiple loggers.
   11:     def self.broadcast(logger) # :nodoc:
   12:       Module.new do
   13:         define_method(:add) do |*args, &block|
   14:           debugger
=> 15:           logger.add(*args, &block) if broadcast_messages
   16:           super(*args, &block)
   17:         end
   18: 
   19:         define_method(:<<) do |x|
(byebug) broadcast_messages
*** NameError Exception: undefined local variable or method `broadcast_messages' for #<RailsStdoutLogging::StdoutLogger:0x005620bd6542d8>

nil
(byebug) logger
#<ActiveSupport::Logger:0x005620be392ad0 @progname=nil, @level=0, @default_formatter=#<Logger::Formatter:0x005620be392a58 @datetime_format=nil>, @formatter=#<ActiveSupport::Logger::SimpleFormatter:0x005620bd6605b0 @datetime_format=nil, @thread_key="activesupport_tagged_logging_tags:47349308261080">, @logdev=#<Logger::LogDevice:0x005620be392a08 @shift_size=nil, @shift_age=nil, @filename=nil, @dev=#<IO:<STDOUT>>, @mutex=#<Logger::LogDevice::LogDeviceMutex:0x005620be3929e0 @mon_owner=nil, @mon_count=0, @mon_mutex=#<Mutex:0x005620be392990>>>, @broadcast_messages=true>

```
just change 

``` ruby 
- logger.add(*args, &block) if broadcast_messages
+ logger.add(*args, &block) if logger.broadcast_messages
```
and all work.

**Comment by grosser** [comment]
FYI fix for rails 4.2

```Ruby
   # show output when running rake, but not duplicate output during console or server
  # https://github.com/rails/rails/pull/22933
  if ENV["RAILS_LOG_TO_STDOUT"] == "true" && defined?(Rake)
    config.logger = ActiveSupport::TaggedLogging.new(ActiveSupport::Logger.new(STDOUT))
  end
```

**Comment by henrytseng** [comment]
Confirmed.
